bin/Analysis.cpp

```diff
@@ -151,6 +155,9 @@ void Analysis::createHistograms() {
 			<< endl;
 	lastNumberOfHistograms = numberOfHistograms;
 
+	pseudoTopAnalyser_->createTrees();
```

Comment:
Within ```while (eventReader->hasNextEvent()) {
initiateEvent();``` you should only have initiated events, hence events that know their data type.
```eventReader->getSeenDatatypes()[DataType::TTJets]``` will give you the same answer.

However, it is different within the createHistograms. Here the treeMan should do the right thing:
```c++
histMan->prepareForSeenDataTypes(eventReader->getSeenDatatypes());
treeMan->prepareForSeenDataTypes(eventReader->getSeenDatatypes());
```
In short:
```c++
if ( eventReader->getSeenDatatypes()[DataType::TTJets] ) {
pseudoTopAnalyser_->createTrees();
unfoldingRecoAnalyser_->createTrees();
}
```
should be managed by the tree manager. It should create trees for these analysers only if the above condition is fulfilled.